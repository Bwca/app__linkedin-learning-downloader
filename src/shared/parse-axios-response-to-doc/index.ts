export { parseAxiosResponseToDoc } from './parse-axios-response-to-doc/parse-axios-response-to-doc';
