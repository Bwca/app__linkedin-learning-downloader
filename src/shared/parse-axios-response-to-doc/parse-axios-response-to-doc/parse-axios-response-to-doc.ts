import { AxiosResponse } from 'axios';
import { JSDOM } from 'jsdom';

export async function parseAxiosResponseToDoc({ data: html }: AxiosResponse): Promise<Document> {
  const {
    window: { document },
  } = new JSDOM(html);

  return document;
}
