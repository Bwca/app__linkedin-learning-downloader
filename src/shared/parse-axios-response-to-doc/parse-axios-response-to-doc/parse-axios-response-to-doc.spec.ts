import { AxiosResponse } from 'axios';

import { parseAxiosResponseToDoc } from './parse-axios-response-to-doc';

describe('parseAxiosResponseToDoc', () => {
  it('should return js Document', async () => {
    // Arrange
    const mockedResponse = { data: mockHtml() } as AxiosResponse;

    // Act
    const doc = await parseAxiosResponseToDoc(mockedResponse);

    // Assert
    expect(doc).toBeTruthy();
    expect(doc).toMatchSnapshot();
  });
});

function mockHtml(): string {
  return `<!DOCTYPE html>
    <html>
    <body>
    
    <h1>My First Heading</h1>
    
    <p>My first paragraph.</p>
    
    </body>
    </html>
    `;
}
