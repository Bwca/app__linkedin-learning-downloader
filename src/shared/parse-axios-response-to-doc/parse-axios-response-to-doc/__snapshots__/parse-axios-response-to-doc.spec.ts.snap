// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`parseAxiosResponseToDoc should return js Document 1`] = `
Document {
  "location": Location {
    "assign": [Function],
    "hash": "",
    "host": "",
    "hostname": "",
    "href": "about:blank",
    "origin": "null",
    "pathname": "blank",
    "port": "",
    "protocol": "about:",
    "reload": [Function],
    "replace": [Function],
    "search": "",
    "toString": [Function],
  },
}
`;
